pkg/sql/opt/optbuilder/testdata/delegate

```diff
@@ -1,17 +1,3 @@
-build
-SHOW DATABASES
```

Comment:
I removed this test because the test catalog does not support accessing virtual schemas other than `information_schema`.

Reply:
cc @RaduBerinde - what was this for? If it was just a smoke test, Rohan maybe you should add another arbitrary `SHOW x` test here (that uses information_schema) to make sure that the delegation is still working as intended

Reply:
There's another show statement below it, so I think thats fine?